extensions/sasl-3.1.md

```diff
@@ -175,7 +202,7 @@ an additional capability.
 
     :server 903 <nick> :SASL authentication successful
 
-`904` aka `ERR_SASLFAIL` is sent when the SASL authentication fails because of invalid credentials or other errors not explicitly mentioned by other numerics.
+`904` aka `ERR_SASLFAIL` is sent when the SASL authentication fails because of invalid credentials or other errors, even if they're additionally sent by other numerics, and must be sent after all additional numerics.
```

Comment:
I would nitpick that because 908 isn't an ERR numeric, it _doesn't_ technically indicate failure, just provide information... On the other hand, how does this change interact with failures which currently _do_ result in an ERR numeric? Should e.g. ERR_TOOLONG or ERR_ABORTED be also followed by ERR_SASLFAIL (they currently aren't)?

Maybe it's better to say "other **error** numerics"?

Reply:
Wow. That is a point that I had COMPLETELY missed. Thank you.

I think that there should then be two RPL and ERR sections to avoid confusion that the RPL_SASLMECHS is not the "ERR" numeric, or at the very least very heavily mention it.

I will have to do some additional research to see if this happens with ERR_TOOLONG and ERR_ABORTED. If that is not the case, then I may revert the change and say that it'll be sent *after* RPL numerics.

Reply:
Resolved in TODO comment (Apr 22 4:40 AM CST)